### ISSUE TITLE
Make producerPaused consistently be a bool

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6792]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Alex reported|
|-|-|
|Trac ID|trac#6792|
|Type|enhancement|
|Created|2013-10-17 13:47:23Z|
|Branch|https://github.com/twisted/twisted/tree/producerPaused-bool-6792|

Right now it's sometimes an int (but not a bool), for what I assume are legacy reasons.

Attachments:

* [t6792.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a7c/a7c40396d44ac088e50afe93f200d4d562f896c1/dffb3165fd73d002a49b5e6605d296dd405fa320.diff) (2535 bytes) - added by Alex on 2013-10-17 13:47:49Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6792 6792
type__enhancement enhancement
reporter__Alex Alex
priority__normal normal
milestone__ 
branch__branches_producerPaused_bool_6792 branches/producerPaused-bool-6792
branch_author__Alex Alex
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1382017643000000 1382017643000000
changetime__1388416856000000 1388416856000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [41052]) Branching to 'producerPaused-bool-6792'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun|
|-|-|

Okay.  Thanks.  Looks pretty good to me.  Two minor tweaks:

  1. Needs a news fragment (but it's a misc so, boring)
  1. `L{bool}` is preferred over `C{bool}` for the type documentation

These are simple enough that I'll go ahead and make them, check buildbot, and merge if it looks good.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [41060]) Merge producerPaused-bool-6792

Author: Alex
Reviewer: exarkun
Fixes: [#6792](https://github.com/twisted/twisted/issues/6792)

Consistently use `bool` instead of `int` to represent whether a registered
producer is paused or not.
